src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_AwaitCompletion.vb

```diff
@@ -0,0 +1,248 @@
+                </Document>)
+                state.SendTypeChars("aw")
+                Await state.AssertSelectedCompletionItem(displayText:="await", isHardSelected:=True, inlineDescription:=CSharpFeaturesResources.Make_container_async)
+
+                state.SendTab()
+                Assert.Equal("
+using System.Threading.Tasks;
+
+public class C
+{
+    public static async Task Main()
+    {
+        await
+    }
+}
+", state.GetDocumentText())
+            End Using
+        End Function
+
+        <WpfFact>
+        Public Async Function AwaitCompletionAddsAsync_LocalFunctionDeclaration() As Task
+            Using state = TestStateFactory.CreateCSharpTestState(
+                <Document><![CDATA[
+using System.Threading.Tasks;
+
+public class C
+{
+    public static Task Main()
+    {
+        Task LocalFunc()
+        {
+            $$
+        }
+    }
+}
+]]>
+                </Document>)
+                state.SendTypeChars("aw")
+                Await state.AssertSelectedCompletionItem(displayText:="await", isHardSelected:=True, inlineDescription:=CSharpFeaturesResources.Make_container_async)
+
+                state.SendTab()
+                Assert.Equal("
+using System.Threading.Tasks;
+
+public class C
+{
+    public static Task Main()
+    {
+        async Task LocalFunc()
+        {
+            await
+        }
+    }
+}
+", state.GetDocumentText())
+            End Using
+        End Function
+
+        <WpfFact>
+        Public Async Function AwaitCompletionAddsAsync_AnonymousMethodExpression() As Task
+            Using state = TestStateFactory.CreateCSharpTestState(
+                <Document><![CDATA[
+using System;
+
+public class C
+{
+    public void F()
+    {
+        Action<int> a = static delegate(int i) { $$ };
+    }
+}
+]]>
+                </Document>)
+                state.SendTypeChars("aw")
+                Await state.AssertSelectedCompletionItem(displayText:="await", isHardSelected:=True, inlineDescription:=String.Empty)
+
+                state.SendTab()
+                Assert.Equal("
+using System;
+
+public class C
+{
+    public void F()
+    {
+        Action<int> a = static delegate(int i) { await };
+    }
+}
+", state.GetDocumentText())
+            End Using
+        End Function
+
+        <WpfFact>
+        Public Async Function AwaitCompletionAddsAsync_SimpleLambdaExpression() As Task
+            Using state = TestStateFactory.CreateCSharpTestState(
+                <Document><![CDATA[
+using System;
+
+public class C
+{
+    public void F()
+    {
+        Action<int> b = static a => { $$ };
+    }
+}
+]]>
+                </Document>)
+                state.SendTypeChars("aw")
+                Await state.AssertSelectedCompletionItem(displayText:="await", isHardSelected:=True, inlineDescription:=String.Empty)
+
+                state.SendTab()
+                Assert.Equal("
+using System;
+
+public class C
+{
+    public void F()
+    {
+        Action<int> b = static a => { await };
```

Comment:
I'd love to have this supported too :)
It can be done as future improvement though.

Reply:
Will look into that as a follow-up.

Reply:
Actually `GetSymbolInfo` returns a symbol for it, but `GetDeclaredSymbol` doesn't. Going to see if this is easy to support in this PR.

Reply:
The corresponding symbol is returned only via `GetSymbolInfo` for this case. So this is getting close. However, this exact test case has a symbol with void return type (since it's an `Action`). The rules for automatically adding `async` applies only for Task-like.

Do we want to tweak things to exclude lambdas from this rule? or just attempt to add a test with `Func<Task>`?